src/rs.cpp

```diff
@@ -1189,47 +1193,67 @@ class on_log_callback : public rs2_log_callback
 void rs2_log_to_callback( rs2_log_severity min_severity, rs2_log_callback_ptr on_log, void * arg, rs2_error** error ) BEGIN_API_CALL
 {
     // Wrap the C function with a callback interface that will get deleted when done
+#if BUILD_EASYLOGGINGPP
     librealsense::log_to_callback( min_severity,
         librealsense::log_callback_ptr{ new on_log_callback( on_log, arg ) }
     );
+#else //BUILD_EASYLOGGINGPP
+    throw std::runtime_error("rs2_log_to_callback is not supported without BUILD_EASYLOGGINGPP");
+#endif //BUILD_EASYLOGGINGPP
 }
 HANDLE_EXCEPTIONS_AND_RETURN( , min_severity, on_log, arg )
 
 
 unsigned rs2_get_log_message_line_number( rs2_log_message const* msg, rs2_error** error ) BEGIN_API_CALL
 {
     VALIDATE_NOT_NULL( msg );
+#if BUILD_EASYLOGGINGPP
     log_message const& wrapper = *(log_message const*) (msg);
     return wrapper.el_msg.line();
+#else //BUILD_EASYLOGGINGPP
+    throw std::runtime_error("rs2_get_log_message_line_number is not supported without BUILD_EASYLOGGINGPP");
+#endif //BUILD_EASYLOGGINGPP
 }
 HANDLE_EXCEPTIONS_AND_RETURN( 0, msg )
 
 const char* rs2_get_log_message_filename( rs2_log_message const* msg, rs2_error** error ) BEGIN_API_CALL
 {
     VALIDATE_NOT_NULL( msg );
+#if BUILD_EASYLOGGINGPP
     log_message const& wrapper = *(log_message const*) (msg);
     return wrapper.el_msg.file().c_str();
+#else //BUILD_EASYLOGGINGPP
+    throw std::runtime_error("rs2_get_log_message_filename is not supported without BUILD_EASYLOGGINGPP");
+#endif //BUILD_EASYLOGGINGPP
 }
 HANDLE_EXCEPTIONS_AND_RETURN( nullptr, msg )
 
 const char* rs2_get_raw_log_message( rs2_log_message const* msg, rs2_error** error ) BEGIN_API_CALL
 {
     VALIDATE_NOT_NULL( msg );
+#if BUILD_EASYLOGGINGPP
     log_message const & wrapper = *( log_message const * )( msg );
     return wrapper.el_msg.message().c_str();
+#else //BUILD_EASYLOGGINGPP
+    throw std::runtime_error("rs2_get_raw_log_message is not supported without BUILD_EASYLOGGINGPP");
+#endif //BUILD_EASYLOGGINGPP
 }
 HANDLE_EXCEPTIONS_AND_RETURN( nullptr, msg )
 
 const char* rs2_get_full_log_message( rs2_log_message const* msg, rs2_error** error ) BEGIN_API_CALL
 {
     VALIDATE_NOT_NULL( msg );
+#if BUILD_EASYLOGGINGPP
```

Comment:
Can we define a log_message API that will centralize these #if statements?

I.e., it would look like:
```
log_message & wrapper = *( log_message * )( msg );
return wrapper.get_full_log_message();
```

And the log_message object (in log.h) would then know to throw if ELPP is not enabled.